service/history/task/transfer_active_task_executor_test.go

```diff
@@ -628,6 +628,45 @@ func (s *transferActiveTaskExecutorSuite) TestProcessDecisionTask_Duplication()
 }
 
 func (s *transferActiveTaskExecutorSuite) TestProcessCloseExecution_HasParent() {
+	s.testProcessCloseExecution_HasParent(
+		s.domainEntry,
+		func(
+			mutableState execution.MutableState,
+			workflowExecution, targetExecution types.WorkflowExecution,
+		) {
+			s.mockVisibilityMgr.On("RecordWorkflowExecutionClosed", mock.Anything, mock.Anything).Return(nil).Once()
+			s.mockArchivalMetadata.On("GetVisibilityConfig").Return(archiver.NewDisabledArchvialConfig())
+		},
+	)
+}
+
+func (s *transferActiveTaskExecutorSuite) TestProcessCloseExecution_HasParentCrossCluster() {
+	s.testProcessCloseExecution_HasParent(
+		s.remoteTargetDomainEntry,
+		func(
+			mutableState execution.MutableState,
+			workflowExecution, targetExecution types.WorkflowExecution,
+		) {
+			s.mockVisibilityMgr.On("RecordWorkflowExecutionClosed", mock.Anything, mock.Anything).Return(nil).Once()
+			s.mockArchivalMetadata.On("GetVisibilityConfig").Return(archiver.NewDisabledArchvialConfig())
+			s.mockClusterMetadata.EXPECT().ClusterNameForFailoverVersion(int64(-24)).Return(s.mockClusterMetadata.GetCurrentClusterName()).AnyTimes()
```

Comment:
We are consistently returning -24 here after the child execution is closed. Not sure where it's coming from

Reply:
-24 is the failover version we used for local domains. https://github.com/uber/cadence/blob/master/common/constants.go#L35